napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,601 @@
+    ----------
+    triangles : np.ndarray
+        (M-2)x3 array of the indices of the vertices that will form the
+        triangles of the triangulation
+    centers : np.ndarray
+        Mx2 array central coordinates of path triangles.
+    offsets : np.ndarray
+        Mx2 array of the offsets to the central coordinates. Offsets need to
+        be scaled by the line width and then added to the centers to
+        generate the actual vertices of the triangulation
+    """
+    for i in range(len(triangles)):
+        ti = triangles[i]
+        p1 = centers[ti[0]] + offsets[ti[0]]
+        p2 = centers[ti[1]] + offsets[ti[1]]
+        p3 = centers[ti[2]] + offsets[ti[2]]
+        if _orientation(p1, p2, p3) < 0:
+            triangles[i] = [ti[2], ti[1], ti[0]]
+
+
+@njit(cache=True, inline='always')
+def _orientation(p1: np.ndarray, p2: np.ndarray, p3: np.ndarray) -> float:
+    """Compute the orientation of three points.
+
+    In terms of napari's preferred coordinate frame (axis 0, y, is pointing
+    down, axis 1, x is pointing right), this orientation is positive for
+    anti-clockwise and negative for clockwise.
+
+    Parameters
+    ----------
+    p1, p2, p3: np.ndarray, shape (2,)
+        The three points to check.
+
+    Returns
+    -------
+    float
+        Positive if anti-clockwise, negative if clockwise, 0 if collinear.
+    """
+    # fmt: off
+    return (
+        (p2[0] - p1[0]) * (p3[1] - p1[1]) - (p2[1] - p1[1]) * (p3[0] - p1[0])
+    )
+    # fmt: on
+
+
+@njit(cache=True, inline='always')
+def _direction_vec_and_length(
+    path: np.ndarray, closed: bool
+) -> tuple[np.ndarray, np.ndarray]:
+    """Calculate the normal vector and its length.
+
+    Parameters
+    ----------
+    path : np.ndarray
+        Mx2 array representing path for which to calculate the direction
+        vectors and the length of each segment.
+        This function assumes that there is no point repetition in the path:
+        consecutive points should not have identical coordinates.
+    closed : bool
+        True if the input path is closed (forms a loop): the edge from the
+        last node in the path back to the first one is automatically inserted
+        if so.
+
+    Returns
+    -------
+    np.ndarray
+        Mx2 array representing containing normal vectors of the path.
+        If closed is False, the last vector has unspecified value.
+    np.ndarray
+        The array of limit of length of inner vectors in bevel joins.
+        To reduce the graphical artifacts, the inner vectors are limited
+        to half of the length of the adjacent egdes in path.
+    """
+    normals = np.empty_like(path)
+    vec_len_arr = np.empty((len(path)), dtype=np.float32)
+    for i in range(1 - closed, len(path)):
+        vec_diff = path[i] - path[i - 1]
+        vec_len_arr[i - 1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[i - 1] = vec_diff / vec_len_arr[i - 1]
+
+    return normals, vec_len_arr * 0.5
```

Comment:
hm. 
Here I return half of vector length

Reply:
Ah good, I was confused about the difference between the variable name and the docs (which I did notice said that the miter was limited to half the length) but I couldn't find where it was. Will update the docs!